### PR TITLE
docs(catalog): add documentation about catalog views

### PR DESCRIPTION
**Pull request by rexledesma**
## Summary & Motivation

Add documentation that this feature exists.

## How I Tested These Changes
N/A

### CODE REVIEW

**Comment by rexledesma** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23318?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#23318** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23318?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#23312** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23312?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rexledesma and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/content/concepts/webserver/ui.mdx [hunk 1]
```diff
@@ -100,6 +100,29 @@ width={1911}
 height={1049}
 />
 
+### Catalog views (Dagster+ Pro)
+
+<Note>A Dagster+ Pro plan is required to use this feature.</Note>
```
Comment (salazarm) [inline docs/content/concepts/webserver/ui.mdx:105]: It's available to all plans

**Review by salazarm — APPROVED** [review]

File: docs/content/concepts/webserver/ui.mdx [hunk 2]
```diff
@@ -100,6 +100,29 @@ width={1911}
 height={1049}
 />
 
+### Catalog views (Dagster+ Pro)
```
Comment (erinkcochran87) [inline docs/content/concepts/webserver/ui.mdx:103]:
```suggestion
</TabItem>
<TabItem name="Catalog views (Dagster+)">

### Catalog views (Dagster+)
```

Let's move this into its own tab so it's easier to find, and also consistent with how other pages in this reference are structured.

**Review by erinkcochran87 — APPROVED** [review]

**Comment by salazarm** [comment]
We should mention that viewers can't create (public, after we change this) catalog views

**Pull request merged**